### PR TITLE
Adds additional documentation for job status

### LINKED ISSUE #68712: API docs for job status should clearly explain what fields indicate success or failure
The API docs for jobs are unclear as to what fields a caller would use to determine job success or failure in 1.11 (and we haven't updated them in a while)

```
  status:
    conditions:
    - lastProbeTime: 2018-09-15T19:51:27Z
      lastTransitionTime: 2018-09-15T19:51:27Z
      message: Job has reached the specified backoff limit
      reason: BackoffLimitExceeded
      status: "True"
      type: Failed
    failed: 6
    startTime: 2018-09-15T19:46:13Z
```

```
$ kubectl explain jobs.status
KIND:     Job
VERSION:  batch/v1

RESOURCE: status <Object>

DESCRIPTION:
     Current status of a job. More info:
     https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status

     JobStatus represents the current state of a Job.

FIELDS:
   active	<integer>
     The number of actively running pods.

   completionTime	<string>
     Represents time when the job was completed. It is not guaranteed to be set
     in happens-before order across separate operations. It is represented in
     RFC3339 form and is in UTC.

   conditions	<[]Object>
     The latest available observations of an object's current state. More info:
     https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/

   failed	<integer>
     The number of pods which reached phase Failed.

   startTime	<string>
     Represents time when the job was acknowledged by the job controller. It is
     not guaranteed to be set in happens-before order across separate
     operations. It is represented in RFC3339 form and is in UTC.

   succeeded	<integer>
     The number of pods which reached phase Succeeded.
```

A casual reader would infer that completionTime is set when the job fails (but it isn't).  Linking out to the docs is ok, but I shouldn't have to click through the links to understand how to use the Job object.  The Job description and the job status fields should *clearly* indicate what fields are set in completion states on the job, because that's 100% of the point of docs.  Taking a summary of the jobs-run-to-completion link and putting it in job.status.conditions, job.status, and job would improve the actual utility of our API doc.

We should also do a pass on other objects and make the actual in server docs more useful.

@kubernetes/sig-apps-bugs

### PR DESCRIPTION
**Pull request by somtochiama**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design

/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Updates API docs for job status

**Which issue(s) this PR fixes**:

Fixes #68712

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/batch/types.go [hunk 1]
```diff
@@ -167,6 +168,7 @@ type JobStatus struct {
 	// Represents time when the job was completed. It is not guaranteed to
 	// be set in happens-before order across separate operations.
 	// It is represented in RFC3339 form and is in UTC.
+	// The completion time is only set when the job finishes successfully
```
Comment (lavalamp) [inline pkg/apis/batch/types.go]: Missing period. (above change, too)

Please make these changes in the public, versioned API, or they won't show up in the documentation.

Comment (SomtochiAma) [reply pkg/apis/batch/types.go]: Done. Are the additional comments sufficient are does more work need to be done?

File: staging/src/k8s.io/api/batch/v1/types.go [hunk 2]
```diff
@@ -131,6 +131,7 @@ type JobSpec struct {
 // JobStatus represents the current state of a Job.
 type JobStatus struct {
 	// The latest available observations of an object's current state.
+	// When a job fails, one of the conditions will have type == "Failed".
```
Comment (lavalamp) [inline staging/src/k8s.io/api/batch/v1/types.go]: Please update both places, even though this is the canonical one :)

Also you'll need to regenerate the generated files. The "verify" test suite is not flaky, if it's failing something needs to be fixed :)

Comment (SomtochiAma) [reply staging/src/k8s.io/api/batch/v1/types.go]: Alright. I will do that now

**Comment by SomtochiAma** [comment]
I think all is good now

**Comment by eloyekunle** [comment]
/lgtm

Ping @lavalamp

**Review by soltysh — APPROVED** [review]
/lgtm
/priority backlog

**Comment by sftim** [comment]
Does any of this extra information also need to go onto https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/?

**Comment by SomtochiAma** [comment]
> Does any of this extra information also need to go onto https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/?

I don't think it does, but I could do it if it required.

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
/approve

Thanks! Documenting this is important.

**Comment by janetkuo** [comment]
/assign @lavalamp

Ask for approval

**Comment by nikhita** [comment]
@SomtochiAma :wave: can you squash the commits into a single commit? Looks ready to go otherwise! :rocket:

**Comment by SomtochiAma** [comment]
> @SomtochiAma 👋 can you squash the commits into a single commit? Looks ready to go otherwise! 🚀

Done! Thank you.

**Comment by nikhita** [comment]
/assign @thockin 
for approval

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**